dateutil/test/test_isoparser.py

```diff
@@ -379,11 +379,19 @@ def __make_date_examples():
 @pytest.mark.parametrize('d,dt_fmt', __make_date_examples())
 @pytest.mark.parametrize('as_bytes', [True, False])
 def test_parse_isodate(d, dt_fmt, as_bytes):
-    d_str = d.strftime(dt_fmt)
-    if isinstance(d_str, six.text_type) and as_bytes:
-        d_str = d_str.encode('ascii')
-    elif isinstance(d_str, bytes) and not as_bytes:
-        d_str = d_str.decode('ascii')
+    if six.PY2:
```

Comment:
I would prefer it if this were written in polyglot style, so nothing that makes explicit reference to `six.PY2`.

That said, this is the wrong thing to do. The original code block was this (I've annotated with comments to make it clear what's happening):

```python
if isinstance(d_str, six.text_type) and as_bytes:
    # If d_str is str/unicode, convert it into bytes
    d_str = d_str.encode('ascii')
elif isinstance(d_str, bytes) and not as_bytes:
    # If d_str is bytes, convert it to str/unicode
    d_str = d_str.decode('ascii')
```

The problem is that I'm expecting to run the same code with `as_bytes=True` and `as_bytes=False`, which means that whatever `d_str` is, it should have to be converted to the other thing half the time. In Python 2 I expect it to be `bytes` and in Python 3 I expect it to be `str`, so *both* of these branches should *already* be hit.

`unicode(b'example')` is the wrong way to convert from bytes to string, the correct way is `b'example'.encode()`, which is what I'm already doing.

Anyway, either something has changed since the original issue was raised or I made a mistake originally, because it seems like these lines are indeed covered, and only the later lines (with `time_val`) are having the coverage problem, so these changes can be removed entirely.